syntax = 'proto3';

package api;

import "k8s.io/api/core/v1/generated.proto";

message JobRequest {
	string Queue = 1;
	string JobSetId = 2;
	double Priority = 3;
	k8s.io.api.core.v1.PodSpec PodSpec = 4;
}

message JobSubmitResponse {
    string JobId = 1;
}

service Submit {
    rpc SubmitJob(JobRequest) returns (JobSubmitResponse) {}
}
